packages/firestore/test/unit/specs/spec_builder.ts

```diff
@@ -936,14 +933,55 @@ export class SpecBuilder {
     }
   }
 
+  private addQueryToActiveTargetsIfNotYet(
+    targetId: number,
+    query: Query,
+    resumeToken?: string
+  ): void {
+    if (this.activeTargets[targetId]) {
+      const specQueriesClone = [...this.activeTargets[targetId].queries];
+      // query is not added to active targets yet.
+      if (
+        !specQueriesClone.some(specQuery =>
+          parseQuery(specQuery).isEqual(query)
+        )
+      ) {
+        specQueriesClone.push(SpecBuilder.queryToSpec(query));
+      }
+      this.activeTargets[targetId] = {
+        queries: specQueriesClone,
+        resumeToken: resumeToken || ''
+      };
+    } else {
+      this.activeTargets[targetId] = {
+        queries: [SpecBuilder.queryToSpec(query)],
+        resumeToken: resumeToken || ''
+      };
+    }
+  }
+
+  private removeQueryFromActiveTargets(query: Query, targetId: number): void {
+    const specQueries = this.activeTargets[targetId].queries.filter(
```

Comment:
As above, this should be inside the if statement.

Reply:
I intend to keep this unchanged, except a better name. This cannot be in if statement, given the way it is written..